### PR TITLE
doc/reduce-traffic: update/clarify max outbound connection count

### LINKED ISSUE #29046: Nit: Inconsistency in the docs regarding block-relay-only connections
I was looking into why my node was frequently opening new block-relay-only connections even though it already had two such nodes connected, and while this appears to be intended behavior to counter eclipse attacks, while searching the codebase for this I encountered a minor contradiction in the docs.

Specifically, [doc/reduce-memory.md](https://github.com/bitcoin/bitcoin/blob/3e691258d8789a4a89cce42e7e71b130491594d7/doc/reduce-memory.md?plain=1#L29) says:

```
- `-maxconnections=<n>` - the maximum number of connections, which defaults to 125. Each active connection takes up some
  memory. This option applies only if inbound connections are enabled; otherwise, the number of connections will not
  be more than 11. Of the 11 outbound peers, there can be 8 full-relay connections, 2 block-relay-only ones,
  and occasionally 1 short-lived feeler or extra outbound block-relay-only connection.
```

While [doc/reduce-traffic.md](https://github.com/bitcoin/bitcoin/blob/3e691258d8789a4a89cce42e7e71b130491594d7/doc/reduce-traffic.md?plain=1#L9) says:

```
By default, Bitcoin Core allows up to 125 connections to different peers, 10 of
which are outbound. You can therefore, have at most 115 inbound connections.
Of the 10 outbound peers, there can be 8 full-relay connections and 2
block-relay-only ones.
```

I'm not sure if the extra feeler/block-relay-only connection takes up one of the maxconnections slots or not, but the latter should probably be updated to reflect this behavior.

### PR DESCRIPTION
**Pull request by MarnixCroes**
closes https://github.com/bitcoin/bitcoin/issues/29046

### CODE REVIEW

**Review by mzumsande — COMMENTED** [review]
Code Review ACK aa4907ae19c797493939dcbf4762cb1666f86546

File: doc/reduce-traffic.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ Reduce Traffic
...
 Some node operators need to deal with bandwidth caps imposed by their ISPs.
 
-By default, Bitcoin Core allows up to 125 connections to different peers, 10 of
-which are outbound. You can therefore, have at most 115 inbound connections.
-Of the 10 outbound peers, there can be 8 full-relay connections and 2
-block-relay-only ones.
+By default, Bitcoin Core allows up to 125 connections to different peers, 11 of
+which are outbound. You can therefore, have at most 114 inbound connections.
+Of the 11 outbound peers, there can be 8 full-relay connections, 2
+block-relay-only ones and occasionally 1 short-lived feeler or an extra block-relay-only connection .
```
Comment (brunoerg) [inline doc/reduce-traffic.md:9]: nit:
```suggestion
block-relay-only ones and occasionally 1 short-lived feeler or an extra block-relay-only connection.
```

Comment (MarnixCroes) [reply doc/reduce-traffic.md]: my bad,
done

**Review by amitiuttarwar — COMMENTED** [review]
ACK d58f89d355faf46b68d0ff8095699d0aff41959c

**Review by brunoerg — APPROVED** [review]
ACK d58f89d355faf46b68d0ff8095699d0aff41959c

**Review by jonatack — COMMENTED** [review]
Post-merge review ACK with logging sanity check

```
2024-01-05T21:54:39.437118Z [Init] m_max_automatic_connections: 125
2024-01-05T21:54:39.437125Z [Init] m_max_outbound_full_relay: 8
2024-01-05T21:54:39.437133Z [Init] m_max_outbound_block_relay: 2
2024-01-05T21:54:39.437139Z [Init] m_max_automatic_outbound: 11
2024-01-05T21:54:39.437145Z [Init] m_max_inbound: 114
```

**Pull request merged**